test/functional/test_framework/test_framework.py

```diff
@@ -251,7 +258,26 @@ def main(self):
         logging.shutdown()
         if cleanup_tree_on_exit:
             shutil.rmtree(self.options.tmpdir)
-        sys.exit(exit_code)
+
+        self.nodes.clear()
+
+        if exit:
+            sys.exit(exit_code)
+
+    def handle_exception(self, e):
```

Comment:
my preference would be to not have this as a separate function from `main()`

Reply:
Addressed in e163c98169e0405255a2e6acdbb679494b5fa42d.